### ISSUE TITLE
21 requests limit

### ISSUE THREAD

**Issue by rodendron** [adw0rd/instagrapi#979]
Hello, friends

Private and Public API, tested various tech accounts (100-200 per each vendor) and various mobile and resident proxies.

graphql requests can make 200-300 and 1200+ requests, but all API parsing (profile, comments) requests (i.instagram) can do only 21 requests. And then i get challenge or logout message. Not 20, not 22 :)) Only 21… => the Logout, Challenge

What kind of magic? I wrote this message after 1 month tests and 20000-30000 requests. Maybe somebody know lifehack? Such as magic request or sequencing? This problem i has from 15 of september.

regards

**Comment by shaklev** [comment]
Any updates on this ? I have a similar scenario ( using Private API ) and range of requests where accounts mostly get blocked is 10-20

**Comment by adw0rd** [comment]
I've only seen something similar in the last week. Some account sellers have a survivability of 20 requests, but others around 100. A week ago there was a survivability of 300-800 requests

**Comment by shaklev** [comment]
@adw0rd Do you consider if the account is connected to a proxy its better if they are from the same location? And also if using mobile proxies what is a good number for the IP pool & good time between IP rotation ?

**Comment by adw0rd** [comment]
@shaklev yes, it is better to have a proxy from one location. By the time of the mobile proxy, I still do not understand how best

**Comment by subzeroid** [comment]
Closing this old support thread as Instagram account/session/rate-limit enforcement rather than a specific instagrapi defect.

These states are usually account, proxy, device, session, action-pattern, or server-side trust decisions. The current guidance is documented here:

- https://subzeroid.github.io/instagrapi/usage-guide/best-practices
- https://subzeroid.github.io/instagrapi/usage-guide/handle_exception
- https://subzeroid.github.io/instagrapi/usage-guide/interactions

If a current release returns a new structured response that should be mapped to a typed exception, please open a fresh focused issue with sanitized `last_json` and the exact method call.
